src/codecatalyst/vue/create/backend.ts

```diff
@@ -57,15 +62,74 @@ export class CodeCatalystCreateWebview extends VueWebview {
         private readonly onComplete: (devenv?: DevEnvironment) => void
     ) {
         super()
+
+        // When webview first loads, an instance of this class
+        // is created.
+        // We build the prompter here since it immeditely starts
+        // fetching the Projects upon creation.
+        // When a user triggers the prompt to select a Project the **first** time,
+        // the fetching of Projects will already be in progress.
+        this.projectPrompter = this.buildPrompter()
     }
 
     public close() {
         this.dispose()
         this.onComplete()
     }
 
-    public async getProjects() {
-        return this.client.listResources('project').flatten().promise()
+    /**
+     * Opens a quick pick that lists all Projects from all Spaces.
+     *
+     * @param spaceName Only show Projects from this Space in the quick pick. If undefined,
+     *                  shows Projects from all Spaces.
+     * @returns Project if it was selected, otherwise undefined due to user cancellation.
+     */
+    public async quickPickProject(spaceName?: CodeCatalystOrg['name']): Promise<CodeCatalystProject | undefined> {
+        // We use an existing prompter since it would have already started
+        // fetching Projects (improved UX).
+        if (this.projectPrompter === undefined) {
+            this.projectPrompter = this.buildPrompter(spaceName)
+        }
+
+        const selectedProject = await this.projectPrompter.prompt()
+        this.projectPrompter = undefined
+
+        if (!isValidResponse(selectedProject)) {
+            return
+        }
+
+        return selectedProject
+    }
+
+    /**
+     * Builds a prompter that allows the user to select a Project.
+     *
+     * @param spaceName Only show Projects from this Space in the quick pick.
+     * @returns prompter without any items in it.
+     */
+    private buildPrompter(spaceName?: CodeCatalystOrg['name']): QuickPickPrompter<CodeCatalystProject> {
+        let projects: AsyncCollection<CodeCatalystProject[]>
+
+        // Determine which Projects from which Spaces need to be fetched.
+        if (spaceName != undefined) {
+            projects = this.client.listProjects({ spaceName: spaceName })
+        } else {
+            projects = this.client.listResources('project')
+        }
+
+        // Convert Projects to quick pick items
+        const projectQuickPickItems = projects.map(projects => {
+            return projects.map(project => ({ label: `${project.org.name}/${project.name}`, data: project }))
+        })
+
+        // Note that the fetching of Projects is **started**
+        // immediately upon **creation** of the prompter.
+        const projectPrompter = createQuickPick<CodeCatalystProject>(projectQuickPickItems, {
+            placeholder: 'Select Space/Project',
+            buttons: [createExitButton()],
+        })
+        projectPrompter.quickPick.title = 'Select Space/Project'
```

Comment:
Can we re-use `createProjectPrompter` here?

Reply:
Good call, was able to get rid of this entire function with this.